### PR TITLE
Add ttl support

### PR DESCRIPTION
**Pull request by yevdev-fiscalnote**

### CODE REVIEW

File: lib/client-transport.js [hunk 1]
```diff
@@ -35,11 +35,13 @@ function TAmqpTransport(client, _options,callback) {
...
-    endpoint:  options.endpoint ||  'amqp://localhost',
-    queueName: options.queueName || 'lycam-rpc',
-    fromName: options.fromName || 'Unknown Client',
-
+  this.options = {
+    endpoint     : options.endpoint || 'amqp://localhost',
+    queueName    : options.queueName || 'lycam-rpc',
+    fromName     : options.fromName || 'Unknown Client',
+    blocking     : (typeof options.messageBlocking === 'undefined' || options.messageBlocking === null) ? options.messageBlocking : true,
+    msg_timeout  : options.messageTtl || 10,
```
Comment (yevdev-fiscalnote) [inline lib/client-transport.js:43]: People got some super computers defaulting to 10ms.

File: lib/client-transport.js [hunk 2]
```diff
@@ -35,11 +35,13 @@ function TAmqpTransport(client, _options,callback) {
...
-  this.options  = {
-    endpoint:  options.endpoint ||  'amqp://localhost',
-    queueName: options.queueName || 'lycam-rpc',
-    fromName: options.fromName || 'Unknown Client',
-
+  this.options = {
+    endpoint     : options.endpoint || 'amqp://localhost',
+    queueName    : options.queueName || 'lycam-rpc',
+    fromName     : options.fromName || 'Unknown Client',
+    blocking     : (typeof options.messageBlocking === 'undefined' || options.messageBlocking === null) ? options.messageBlocking : true,
```
Comment (pater24) [inline lib/client-transport.js]: so when you pass this in, is it "false"?

**Review by pater24 — COMMENTED** [review]
Just 1 question

File: lib/thrift-amqp.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ exports.createConnection = function (options) {
 		return conn;
 };
 
-exports.createClient = function (ServiceClient, connection) {
+exports.createClient = function (ServiceClient, connection, transportOptions) {
```
Comment (pater24) [inline lib/thrift-amqp.js]: should this be optional?

**Review by pater24 — APPROVED** [review]

**Pull request merged**